test.c

```diff
@@ -1107,7 +1107,13 @@ static void test_blocking_connection_timeouts(struct config config) {
     test("Does not return a reply when the command times out: ");
     if (detect_debug_sleep(c)) {
         redisAppendFormattedCommand(c, sleep_cmd, strlen(sleep_cmd));
+
+        // flush connection buffer without waiting for the reply
```

Comment:
An alternative could be to use `redisBufferWrite()` instead, which resets the `obuf`